### ISSUE TITLE
off_t data type is unknown

### ISSUE THREAD

**Issue by phmarek** [philburk/pforth#11]
First error is

```
c99 -DPF_EMBEDDED=1 -I. -DPF_SUPPORT_FP -D_GNU_SOURCE -c -o pf_cglue.o ../../csrc/pf_cglue.c
In file included from ../../csrc/pf_all.h:50:0,
                 from ../../csrc/pf_cglue.c:19:
../../csrc/pf_io.h:87:42: error: unknown type name ‘off_t’
```

but I guess there'll be some more.

**Comment by philburk** [comment]
What target are you building for?
What host are you building on?

**Comment by phmarek** [comment]
That's on Debian testing/unstable, amd64; not cross-compiling.

**Comment by philburk** [comment]
The use of off_t was added in this commit:
https://github.com/philburk/pforth/commit/336369a5b94ae6588fb891606d37053f9fb8c2d6

Because off_t is not very standard, I recommend replacing off_t with a new typedef that is based on int64_t.

Forth uses two cells to hold a seek offset, which will either be 64 or 128-bits in pForth.
We don't really need to go to 128-bit offsets so internally we can just always use 64.

**Comment by philburk** [comment]
To test the FILE words affected by this change.

    s" ../../readme.txt"  r/o open-file . value FID

That should print 0

    100 s>d fid reposition-file
    fid file-position

Should return a position of 100

**Comment by philburk** [comment]
I decided to define file_offset_t as long because it is more compatible with stdout.

Pending fix here:
https://github.com/philburk/pforth/pull/44

**Comment by philburk** [comment]
This was fixed by #44
